src/NServiceBus.Core/Reliability/Outbox/OutboxTransaction.cs

```diff
@@ -7,7 +7,7 @@ namespace NServiceBus.Outbox;
 /// <summary>
 /// Transaction in which storage operations must enlist to be consistent with the outbox operations.
 /// </summary>
-public interface IOutboxTransaction : IDisposable
+public interface IOutboxTransaction : IDisposable, IAsyncDisposable
```

Comment:
This change will likely result in broken downstreams but as we're working on master and that is v10 I would be fine with merging this in a breaking way.